.vscode/extensions.json

```diff
@@ -0,0 +1,6 @@
+{
+    "recommendations": [
+        "ms-vscode.cpptools",
+        "ms-vscode.cmake-tools"
```

Comment:
did you tests if this extension could be used without the changes to the CMakefiles and build.py?

Reply:
It actually seems that more recent versions of the extension can bypass the old problem of the CMake cache loop, which makes it less useful, but it looks like it still has the IntelliSense problems. With these changes IntelliSense works properly. 

Without change:
![image](https://user-images.githubusercontent.com/36864441/108549836-a1a5b180-72e5-11eb-8586-3af434ef5bf9.png)

With change:
![image](https://user-images.githubusercontent.com/36864441/108549894-b41feb00-72e5-11eb-8723-f60f96b00216.png)

It also makes it so that the compilers that the user specifies that they want to use are actually used (if this wasn't in place, then they could set their compiler and it is just completely ignored because of the manual include in the toolchain). It gives the user better flexibility in my opinion.